app/src/main/java/fr/neamar/kiss/DataHandler.java

```diff
@@ -199,6 +214,46 @@ public void addToHistory(Context context, String id) {
         DBHelper.insertHistory(context, currentQuery, id);
     }
 
+    public void addShortcut(Context context, ShortcutPojo pojo) {
+        ShortcutRecord record = new ShortcutRecord();
+        record.name = pojo.name;
+        record.iconResource = pojo.resourceName;
+        record.packageName = pojo.packageName;
+        record.intentUri = pojo.intentUri;
+        
+        if (pojo.icon != null) {
+            ByteBuffer bb = ByteBuffer.allocate(pojo.icon.getRowBytes() * pojo.icon.getHeight());        
+            pojo.icon.copyPixelsToBuffer(bb);        
+            record.icon_blob = bb.array();
+        }
+
+
+        DBHelper.insertShortcut(context, record);
+    }
+
+    public ArrayList<ShortcutPojo> getShortcuts(Context context) {
```

Comment:
same thing: this can be done in doInBackground on LoadShortcutPojo (same method that's used on other Loaders)